src/core/lib/iomgr/ev_poll_posix.cc

```diff
@@ -1379,7 +1379,9 @@ static void reset_event_manager_on_fork() {
   gpr_mu_lock(&fork_fd_list_mu);
   while (fork_fd_list_head != nullptr) {
     if (fork_fd_list_head->fd != nullptr) {
-      close(fork_fd_list_head->fd->fd);
+      if (!fork_fd_list_head->fd->closed) {
+        close(fork_fd_list_head->fd->fd);
```

Comment:
Shouldn't fork_fd_list_head->fd->closed be set after closing the fd?

Reply:
Good catch! I thought if the linked list is depleted then maybe no one cares about the state of the `grpc_fd`, but I'm not 100% sure if other portion of code won't use those state after forking.

I created another patch at https://github.com/grpc/grpc/pull/20893. But I not sure about if we have sufficient test coverage for behaviors after forking.

CC @ericgribkoff

Reply:
One more point, the `closed` flag is not flipped before, so this change should be strictly better than before, while it doesn't guarantee to solve other existing problem.